### PR TITLE
feat: Updates to the alignment entity

### PR DESCRIPTION
**Pull request by manasaV3**
Relates to

### Changes to the config schema:
- Updates the `is_canonical` -> `is_portal_standard`
- Adds the `method_type`
- Updates the spelling of volume_dimension

### Changes to the configs:
- Adds migration handler for `alignment.method_type` in the config. It follows the following conditions:
```
Everything that is FIDUCIAL is fiducial_based
Everything that mentions AreTomo as reconstruction software is projection matching
If reconstruction_software=IMOD and NON_FIDUCIAL is patch_tracking
Else undefined
```
- Removes trailling whitespaces in the configs

### S3 Workflows:
- Adds "files" field to the alignment metadata. This lists all the files associated to this alignment
- Adds `get_name_and_path` class method to the AlignmentImporter. This allows the DestinationFilterFinder to reference an existing Alignment.

### CODE REVIEW

File: ingestion_tools/dataset_configs/template.yaml [hunk 1]
```diff
@@ -1,4 +1,25 @@
 # Future TODO: Create example_config.yaml file to show how to use this template.
+alignments: OPTIONAL
+- metadata: OPTIONAL
+    alignment_type: OPTIONAL ENUM (LOCAL, GLOBAL)
+    format: OPTIONAL ENUM (IMOD, ARETOMO)
+    method_type: OPTIONAL ENUM (fiducial_based, patch_tracking, projection_matching)
+    is_portal_standard: OPTIONAL, BOOLEAN (DEFAULT FALSE)
+    offset: OPTIONAL
```
Comment (Bento007) [inline ingestion_tools/dataset_configs/template.yaml:8]: thanks for adding these

File: ingestion_tools/scripts/schema_migration/upgrade.py [hunk 2]
```diff
@@ -34,7 +36,7 @@ def upgrade_config(data: dict[str, Any]) -> dict[str, Any]:
 
 
 def has_changes(file, config):
-    with open(file, 'r') as file:
```
Comment (Bento007) [inline ingestion_tools/scripts/schema_migration/upgrade.py:37]: I think our formatters disagree

File: ingestion_tools/scripts/importers/alignment.py [hunk 3]
```diff
@@ -154,13 +160,13 @@ def get_default_metadata(cls) -> dict[str, Any]:
...
             "x_rotation_offset": 0,
         }
 
-    def get_tiltseries_path(self) -> str | None:
-        for ts in TiltSeriesImporter.finder(self.config, **self.parents):
-            return ts.get_metadata_path()
-        return None
+    @classmethod
+    def get_name_and_path(cls, metadata: dict, name: str, path: str, results: dict[str, str]) -> [str, str, dict]:
+        paths = {filname: filname for filname in metadata.get("files", [])}
```
Comment (jgadling) [inline ingestion_tools/scripts/importers/alignment.py:171]: typo: `filename` vs `filname`

File: schema/core/v1.1.0/metadata.yaml [hunk 4]
```diff
@@ -817,7 +817,7 @@ classes:
       offset:
         description: *desc_alignment_offset
         range: AlignmentOffset
-      volume_dimesion:
+      volume_dimension:
```
Comment (Bento007) [inline schema/core/v1.1.0/metadata.yaml:820]: good catch

**Review by Bento007 — APPROVED** [review]

**Review by jgadling — APPROVED** [review]
LGTM after the typo is fixed

File: ingestion_tools/scripts/schema_migration/upgrade.py (hunk 2 above)
Comment (manasaV3) [reply ingestion_tools/scripts/schema_migration/upgrade.py:37]: wdym? 😅

**Pull request merged**